File: schema/address.go
package schema

import (
	"github.com/graphql-go/graphql"
)

var AddressType = graphql.NewObject(graphql.ObjectConfig{
	Name: "Address",
	Fields: graphql.Fields{
		"id": &graphql.Field{
			Type: graphql.NewNonNull(graphql.Int),
		},
		"name": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
		"line1": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
		"line2": &graphql.Field{
			Type: graphql.String,
		},
		"line3": &graphql.Field{
			Type: graphql.String,
		},
		"city": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
		"region": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
		"postalCode": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
		"country": &graphql.Field{
			Type: graphql.NewNonNull(graphql.String),
		},
	},
})

var AddressInputSchema = graphql.NewInputObject(graphql.InputObjectConfig{
	Name: "AddressInput",
	Fields: graphql.InputObjectConfigFieldMap{
		"name": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
		"line1": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
		"line2": &graphql.InputObjectFieldConfig{
			Type: graphql.String,
		},
		"line3": &graphql.InputObjectFieldConfig{
			Type: graphql.String,
		},
		"city": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
		"region": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
		"postalCode": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
		"country": &graphql.InputObjectFieldConfig{
			Type: graphql.NewNonNull(graphql.String),
		},
	},
})

File: frontend/src/components/Address.js
import React from "react";
import { gql } from "apollo-boost";

export default function Address({ address }) {
  return (
    <address>
      <strong>{address.name}</strong> <br />
      {address.line1} <br />
      {address.line2 && (
        <React.Fragment>
          {address.line2} <br />
        </React.Fragment>
      )}
      {address.line3 && (
        <React.Fragment>
          {address.line3} <br />
        </React.Fragment>
      )}
      {address.city}, {address.region}, {address.postalCode} <br />
      {address.country}
    </address>
  );
}

Address.fragments = {
  address: gql`
    fragment AddressAddress on Address {
      name
      line1
      line2
      line3
      city
      region
      postalCode
      country
    }
  `
};
